### PR TITLE
Feature/create target

### PR DESCRIPTION
**Pull request by brunoparma88**
As a user I should be able to create a new target so I can find people with similar interests in a specific area

### CODE REVIEW

File: app/controllers/api/v1/targets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+module Api
+  module V1
+    class TargetsController < Api::V1::ApiController
+      helper_method :target
+
+      def create
+        @target = current_user.targets.create! targets_params
```
Comment (jpascual1994) [inline app/controllers/api/v1/targets_controller.rb]:
```suggestion
        @target = current_user.targets.create!(targets_params)
```

File: app/controllers/api/v1/targets_controller.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module Api
+  module V1
+    class TargetsController < Api::V1::ApiController
+      helper_method :target
```
Comment (jpascual1994) [inline app/controllers/api/v1/targets_controller.rb]: Where it's defined?

File: spec/request/api/v1/targets/create_spec.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+describe 'POST api/v1/targets/', type: :request do
+  subject do
+    post api_v1_targets_path, params: params, headers: auth_headers, as: :json
+    response
+  end
+
+  let(:user)           { create :user }
```
Comment (jpascual1994) [inline spec/request/api/v1/targets/create_spec.rb]: Please remove the extra empty spaces

File: spec/request/api/v1/targets/create_spec.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    end
+
+    it 'creates the target' do
+      expect {
+        subject
+      }.to change(Target, :count).by(1)
+    end
+  end
+
+  context 'with invalid params' do
```
Comment (jpascual1994) [inline spec/request/api/v1/targets/create_spec.rb:43]: Maybe you can also add a test case for a request with no user logged in, it should return unauthorized

File: config/application.rb [hunk 5]
```diff
@@ -19,7 +19,7 @@ class Application < Rails::Application
     #
     # config.time_zone = "Central Time (US & Canada)"
     # config.eager_load_paths << Rails.root.join("extras")
-
+    #
```
Comment (martinjaimem) [inline config/application.rb]: not needed

File: spec/models/target_spec.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
```
Comment (martinjaimem) [inline spec/models/target_spec.rb:2]: not needed 👍

File: spec/models/target_spec.rb [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+describe Target do
+  describe 'Associations' do
+    it { should belong_to(:user) }
```
Comment (martinjaimem) [inline spec/models/target_spec.rb]:
```suggestion
    it { is_expected.to belong_to(:user) }
```

same for all 👍

File: spec/request/api/v1/targets/create_spec.rb [hunk 8]
```diff
@@ -0,0 +1,71 @@
+require 'rails_helper'
+
```
Comment (martinjaimem) [inline spec/request/api/v1/targets/create_spec.rb]:
```suggestion
```

File: spec/request/api/v1/targets/create_spec.rb [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+describe 'POST api/v1/targets/', type: :request do
+  let(:user) { create :user }
```
Comment (martinjaimem) [inline spec/request/api/v1/targets/create_spec.rb]:
```suggestion
  let(:user) { create(:user) }
```

File: spec/request/api/v1/targets/create_spec.rb [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+  let(:topic) { create :topic }
```
Comment (martinjaimem) [inline spec/request/api/v1/targets/create_spec.rb]:
```suggestion
  let(:topic) { create(:topic) }
```

File: spec/request/api/v1/targets/create_spec.rb [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+          radius: 150
+        }
+      }
+    end
+
+    it 'returns success' do
+      expect(subject).to have_http_status(:success)
+    end
+
+    it 'creates the target' do
```
Comment (martinjaimem) [inline spec/request/api/v1/targets/create_spec.rb:36]: what about the title/long/lat ... ?

File: spec/request/api/v1/targets/create_spec.rb [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+    end
+  end
+
+  context 'with no user logged in' do
+    subject do
+      post api_v1_targets_path, params: {}, as: :json
+      response
+    end
+
+    it 'returns the missing params error' do
```
Comment (martinjaimem) [inline spec/request/api/v1/targets/create_spec.rb]: as a general rule, the tests inside not auth should test the response and that the action was not performed (in this case, record not created)

File: spec/routing/target_routing_spec.rb [hunk 13]
```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+
```
Comment (martinjaimem) [inline spec/routing/target_routing_spec.rb]:
```suggestion
```

File: spec/routing/target_routing_spec.rb [hunk 14]
```diff
@@ -0,0 +1,9 @@
...
+describe Api::V1::TargetsController, type: :routing do
+  describe 'routing' do
+    it 'routes to #create' do
+      expect(POST: '/api/v1/targets').to route_to('api/v1/targets#create', format: :json)
+    end
+  end
+end
```
Comment (martinjaimem) [inline spec/routing/target_routing_spec.rb:7]: Cooooooool 🚀

File: app/models/topic.rb [hunk 15]
```diff
@@ -13,5 +13,6 @@
 #
 class Topic < ApplicationRecord
   validates :name, presence: true, uniqueness: true
+  has_many :targets, dependent: :destroy
```
Comment (martinjaimem) [inline app/models/topic.rb:16]: what about adding a test for this?

**Review by jpascual1994 — APPROVED** [review]

File: spec/request/api/v1/targets/create_spec.rb [hunk 16]
```diff
@@ -0,0 +1,86 @@
+require 'rails_helper'
```
Comment (martinjaimem) [inline spec/request/api/v1/targets/create_spec.rb]: not needed 😄

**Review by martinjaimem — APPROVED** [review]
LGTM

**Pull request merged**